cmd/metal-api/internal/service/machine-service.go

```diff
@@ -591,8 +592,10 @@ func (r machineResource) registerMachine(request *restful.Request, response *res
 			Allocation:  nil,
 			SizeID:      size.ID,
 			PartitionID: partition.ID,
-			RackID:      requestPayload.RackID,
-			Hardware:    machineHardware,
+			// the RackID is calculated by the switch connections
+			// a metal-core instance might respond to pxe requests from all racks
+			//RackID:      requestPayload.RackID,
```

Comment:
Field should probably be deprecated / removed in the machine register request.

Reply:
Good point. I deprecated the field.